addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1148 @@
+        return self.write({
+            'next_activity_id': False,
+            'date_action': False,
+            'title_action': False,
+        })
+
+    # ----------------------------------------
+    # Mail Gateway
+    # ----------------------------------------
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if 'stage_id' in init_values and self.probability == 100 and self.stage_id.on_change:
+            return 'crm.mt_lead_won'
+        elif 'active' in init_values and self.probability == 0 and not self.active:
+            return 'crm.mt_lead_lost'
+        elif 'stage_id' in init_values and self.stage_id and self.stage_id.sequence <= 1:
+            return 'crm.mt_lead_create'
+        elif 'stage_id' in init_values:
+            return 'crm.mt_lead_stage'
+        return super(CrmLead, self)._track_subtype(init_values)
+
+    @api.multi
+    def _notification_group_recipients(self, message, recipients, done_ids, group_data):
+        """ Override the mail.thread method to handle salesman recipients.
+        Indeed those will have specific action in their notification emails. """
+        group_sale_salesman = self.env.ref('base.group_sale_salesman').id
+        for recipient in recipients:
+            if recipient.id in done_ids:
+                continue
+            if recipient.user_ids and group_sale_salesman in recipient.user_ids[0].groups_id.ids:
+                group_data['group_sale_salesman'] |= recipient
+                done_ids.add(recipient.id)
+        return super(CrmLead, self)._notification_group_recipients(message, recipients, done_ids, group_data)
+
+    @api.multi
+    def _notification_get_recipient_groups(self, message, recipients):
+        res = super(CrmLead, self)._notification_get_recipient_groups(message, recipients)
+
+        won_action = self._notification_link_helper('method', method='case_mark_won')
+        lost_action = self._notification_link_helper('method', method='case_mark_lost')
+        convert_action = self._notification_link_helper('method', method='convert_opportunity')
+
+        if self.lead_type == 'lead':
+            res['group_sale_salesman'] = {
+                'actions': [{'url': convert_action, 'title': 'Convert to opportunity'}]
+            }
+        else:
+            res['group_sale_salesman'] = {
+                'actions': [
+                    {'url': won_action, 'title': 'Won'},
+                    {'url': lost_action, 'title': 'Lost'}]
+            }
+        return res
+
+    @api.model
+    def message_get_reply_to(self, res_ids, default=None):
+        """ Override to get the reply_to of the parent project. """
+        leads = self.sudo().browse(res_ids)
+        team_ids = set(leads.mapped('team_id').ids)
+        aliases = self.env['crm.team'].message_get_reply_to(list(team_ids), default=default)
+        return dict((lead.id, aliases.get(lead.team_id and lead.team_id.id or 0, False)) for lead in leads)
```

Comment:
aliases.get(lead.team_id and lead.team_id.id or 0, False) ==>  aliases.get(lead.team_id.id) should work.